### DISCUSSION TITLE
Search indexing CRON URL improvements

### DISCUSSION THREAD

**Discussion by Caleydon** [PrestaShop/PrestaShop#40207] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

The PrestaShop using CRON URL in order to re-build searching index automatically (BO area: Admin > Shop Parameters > Search). But this CRON URL has including name of the admin folder and it is make any issues I listed below.

Here is the example of this CRON URL:

`https://www.mydomain.com/admin123/index.php?controller=AdminSearch&action=searchCron&ajax=1&full=1&token=5MFe28qx&id_shop=1`

### Describe the solution you'd like

The issues within this CRON URL:

1. admin folder name is (almost) private parameter and individual choice of the PrestaShop user - you made this feature by security reasons in the past. I guess such parameter should not be included in the URL such CRON, because the URL is added into CRON manager (webhosting service) outside of the PrestaShop admin. So everybody who have an access into CRON manager the easy will know URL into merchant's back office login.

2. the main issue: CRON URL is added into CRON task manager on the web hosting. When name of the admin folder will be changed e.g. by security reasons, CRON URL stop working and need to be immediately changed in the CRON task manager. But I saw multiple times that merchant didn't know about needing modifying CRON URL after renaming admin folder. Nobody and nothing warning them about this needing. This obligation know PrestaShop developers and true insiders. 

The merchants very often contact me with an issue "My searching is broken - my visitors can't find all offered products by searching keywords..." 

So I opened this feature request and ask to:

1. please rewamp this CRON URL mechanism and exclude admin folder name. 

2. it could be very nice to have (in the Admin > Shop Parameters > Search) timestamp of the most recent CRON run because nobody know if the CRON working automatically or not without such information.

3. please make a significant global back office warning “Action needed” (visible for example everywhere under the BO header) when the search index is not complete and number of available and indexed products are not equal. I guess it is a quite critical information for search feature and it shouldn't hide to merchant and ask an action.

4. often I solved a problem with search indexing where database tables "PREFIX_search_index" and "PREFIX_search_word" were corrupted because of very large product catalog. By reseting content of these tables via database manager helps resolve such type of issue. But maybe could be helpful have an button "Clean & Reset Search index" directly in BO that completelly repair tables by following queries:
```
TRUNCATE TABLE `ps_search_index`;
TRUNCATE TABLE `ps_search_word`;
```

![ps-search-feature](https://user-images.githubusercontent.com/2806752/164889642-d9c5378c-f237-4a22-9d89-94e70affbd6e.png)

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by hibatallahAouadni** [comment]
Hello @Caleydon 

Thank you for your suggestion and all detailed specifictaions :clap:  
The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you
